src/contextbuilder.cpp

```diff
@@ -298,13 +321,23 @@ M2k *ContextBuilder::m2kOpen()
 
 void ContextBuilder::contextClose(Context* device, bool deinit)
 {
+	std::string uri = device->getUri();
+	bool isDeviceCached = reference_count.find(uri) != reference_count.end();
+	if(isDeviceCached) {
+		reference_count[uri]--;
```

Comment:
You have a function to increment, but you don't have one for decrement. Maybe review this